scripts/generate_config_tests.py

```diff
@@ -58,14 +58,16 @@ def single_setting_case(setting: config_common.Setting, when_on: bool,
     'MBEDTLS_ERROR_STRERROR_DUMMY': '!MBEDTLS_ERROR_C',
     'MBEDTLS_GENPRIME': 'MBEDTLS_RSA_C',
     'MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES': 'MBEDTLS_ENTROPY_C',
-    'MBEDTLS_NO_PLATFORM_ENTROPY': 'MBEDTLS_ENTROPY_C',
     'MBEDTLS_PKCS1_V15': 'MBEDTLS_RSA_C',
     'MBEDTLS_PKCS1_V21': 'MBEDTLS_RSA_C',
     'MBEDTLS_PSA_CRYPTO_CLIENT': '!MBEDTLS_PSA_CRYPTO_C',
     'MBEDTLS_PSA_INJECT_ENTROPY': 'MBEDTLS_PSA_CRYPTO_C',
     'MBEDTLS_PSA_ASSUME_EXCLUSIVE_BUFFERS': 'MBEDTLS_PSA_CRYPTO_C',
 }
 
+if build_tree.is_mbedtls_3_6():
+    SIMPLE_DEPENDENCIES['MBEDTLS_NO_PLATFORM_ENTROPY'] = 'MBEDTLS_ENTROPY_C'
```

Comment:
`SIMPLE_DEPENDENCIES` and similar data should probably be in the consuming branch (like the exceptions in `analyze_outcomes.py`). Out of scope here. Do we already have a task to track this @mpg ?